comictalker/comictalkerapi.py

```diff
@@ -16,23 +16,36 @@
 from __future__ import annotations
 
 import logging
+import pathlib
+from collections.abc import Mapping
+from typing import Any
 
 import comictalker.talkers.comicvine
 from comictalker.talkerbase import ComicTalker, TalkerError
 
 logger = logging.getLogger(__name__)
 
 
-def get_comic_talker(source_name: str) -> type[ComicTalker]:
-    """Retrieve the available sources modules"""
-    sources = get_talkers()
-    if source_name not in sources:
-        raise TalkerError(source=source_name, code=4, desc="The talker does not exist")
-
-    talker = sources[source_name]
-    return talker
-
-
-def get_talkers() -> dict[str, type[ComicTalker]]:
-    """Returns all comic talker modules NOT objects"""
-    return {"comicvine": comictalker.talkers.comicvine.ComicVineTalker}
+def set_talker_settings(talker: ComicTalker, settings: dict[str, Any]) -> None:
+    try:
+        talker.parse_settings(settings)
+    except Exception as e:
+        logger.exception(
+            f"Failed to set talker settings for {talker.source_details.name}, will use defaults. Error: {str(e)}",
+        )
+        raise TalkerError(source=talker.source_details.name, code=4, desc="Could not apply talker settings")
```

Comment:
What's the point of this function?
Currently the only place this is used does this
```python
try:
    for talker_name, talker in self.talker_plugins.items():
        ct_api.set_talker_settings(talker, self.options[0][talker_name])
except Exception as e:
    # Remove talker as we failed to apply the settings
    del self.talker_plugins[e.source]  # type: ignore[attr-defined]
    logger.exception("Failed to initialize talker settings. Error %s", str(e))
```
Which does not benefit from anything that this function does and catches the wrong exceptions and only removes the first talker that fails.

It should be:
```python
for talker_name, talker in list(self.talker_plugins.items()):
    try:
        ct_api.set_talker_settings(talker, self.options[0][talker_name])
    except ct_api.TalkerError as e:
        # Remove talker as we failed to apply the settings
        del self.talker_plugins[e.source]
        logger.exception("Failed to initialize talker settings. Error %s", e)
```

Which can entirely be replaced with this ignoring the `set_talker_settings` function:
```python
for talker_name, talker in list(self.talker_plugins.items()):
    try:
        talker.parse_settings(self.options[0][talker_name])
    except Exception as e:
        # Remove talker as we failed to apply the settings
        del self.talker_plugins[talker_name]
        logger.exception("Failed to initialize talker settings. Error %s", e)
```

Reply:
I think it's where it was leftover and I didn't think it through.